Assess OTP2 upgrade

rachelekm opened the pull request:
## Overview

Assesses upgrade to OpenTripPlanner 2 and compile notes into decision ADR. Decision summary: OTP2 does not have support for core application features so we will not be moving forward with the major version upgrade, although there is a minor version upgrade for OTP1 that is the "final release" and includes backported bugfixes.


### Notes

I started building a diagram to better understand how this project utilizes trip planning requests to OTP and its graph-building process, but it may adjust as I get deeper into #1385 so I will include that finished diagram as part of that PR as a README in this new `doc` directory.

Connects #1355

rajadain commented on doc/adr-001-otp2-update-eval.md:
```diff
@@ -0,0 +1,106 @@
+- Does not support core features
+    - OTP2 has removed isochrone generation entirely, so we will not be able to generate the travel shed geometry which we use to find destinations within a defined travel radius. This critically affects the “Explore” mode
+- Sweeping changes to routing API
+    - Does not support "bcycle" rental bike updator out of the box, which we currently use for indigo rental bike information like station updates
```
Is `bcycle` a typo?

rachelekm replied:
It's actually the name of the manufacturers behind Indego bikes and stations (and other cities' bike shares)! It seems like they[ maintain the GBFS spec](https://www.bcycle.com/gbfs) and then the operator (indigo) opts in, so we configure the otp server with our "bcycle indigo" url that pings for bike share updates.

rajadain reviewed (changes requested):
Good reasoning and recap. I agree, we should go with OTP 1.5 for now, especially given the short-ish expected lifespan of this project. Will approve after syntax edits are made.

rachelekm commented:
> Good reasoning and recap. I agree, we should go with OTP 1.5 for now, especially given the short-ish expected lifespan of this project. Will approve after syntax edits are made.

@rajadain , thank you for the review! I just addressed those edits, this is ready for another look. Noting that I rebased with develop following the recent monthly release.

rajadain commented on doc/adr-001-otp2-update-eval.md:
```diff
@@ -0,0 +1,98 @@
+- Aligned with GTFS and GBFS feed formats and likely will maintain alignment
+- More efficient graph building and performant routing 
+
+**Cons**
```
```suggestion
**Cons:**
```

rajadain reviewed (approved):
Nicely done!

rachelekm commented:
Thank you for the review! I captured that last grammatical fix in the latest commit following rebase.
